OADP-2257 CloudStorage: Cannot Restore CSI backups in different AWS region (#67918)
openshift/openshift-docs: modules/oadp-installing-oadp-rosa-sts.adoc

[IMPORTANT]
====

> Restic and Kopia are not supported in the OADP on ROSA with AWS STS environment. Make sure that the Restic/Kopia node agent is disabled. Use native snapshots to back up volumes. See _Known Issues_ for more information.

Reviewer: Use native snapshots or CSI snapshot

====

[IMPORTANT]
====

> In an Amazon ROSA cluster using STS authentication, restoring backed-up data in a different AWS region is not supported.

Reviewer 2: ```suggestion
ROSA clusters with STS do not support moving backed-up data across AWS regions. 

```

Reviewer 2: Hi @sbeskin-redhat,
   Hope all is well! Looking at the JIRA ticket that supports this PR,  I am not sure if what is being said is whether the data can be backed up across regions but then can't be restored, OR it can't be backed up at all. Seems that if it can't be restored from a different region then where the cluster is housed, then it shouldnt be able to be backed up to a different region in the first place? Sorry if I am making it more confusing then it needs to be!

Reviewer 3: IIRC we're saying if your credential is STS authentication, the token used can only accessed (or at least in a way that we support) resources in that region. And so we can only upload backed up data to that region's object storage.

If you want to restore from another region, you have to on your own (which we are providing support) find a way to move data from objectStorageRegionA to objectStorageRegionB and use another regionB_STS_Credentials to restore.

Author: @mletalie 
"...do not support moving backed-up data across AWS regions." - this is not exactly correct. Data can be moved across regions using native AWS tools. But OADP does not support restoring backed-up data in an AWS region different from the one in which it was backed up.  Data can only be backed up and then restored within the same AWS region.

> The Data Mover feature is not currently supported in ROSA clusters. You can use native AWS S3 tools for moving data.

Reviewer 2: IS this feature not available in all ROSA clusters (ie with vs w/out STS)? Not sure. 
Also, b/c we are telling the reader about an option that they can use for back-up/restoration of data purposes, should we have a link to that info? Given that there is the AWS S3 tools for that purpose, do we need to even tell them about the Data Mover feature? Again, not sure but maybe something to consider.

Reviewer 3: I think this is limited to ROSA STS.

Author: @mletalie 
- STS is the recommended credential mode for ROSA clusters, so we are not talking about ROSA clusters without STS here. 
- The option that they can use for back-up/restoration of data purposes is OADP, and there is a link to the OADP documentation.
- The AWS S3 tools that are mentioned here are for moving data, not for  backup/restore. They are mentioned because the usual OCP tools such as Data Mover and OADP backup/restore across regions are not supported here.

Restic is not supported in the OADP on ROSA with AWS STS environment. Ensure the Restic service is disabled. Use native snapshots to backup volumes. See _Known Issues_ for more information.
====

[IMPORTANT]
====

> In an Amazon ROSA cluster using STS authentication, you cannot restore your backed-up data in a different AWS region.

Reviewer 3: ```suggestion
In an Amazon ROSA cluster using STS authentication, restoring backed-up data in a different AWS region is not supported.
```
